spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -605,7 +605,8 @@ public static void importSparkPartitions(
           filesToImport
               .map((MapFunction<DataFile, String>) f -> f.path().toString(), Encoders.STRING())
```

Comment:
you may want to add this change here as well for unpartitioned tables
https://github.com/apache/iceberg/blob/84e390c5ca541bf85a9c29b4201443d7497fe17f/spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java#L536